### PR TITLE
[buck] enable junit5

### PR DESCRIPTION
**Pull request by scottlee14**
Enables junit5 on addebuck. This is accomplished by
- pulling in a bunch of junit5 dependencies. Buck version controls all third-party dependencies, so they needed to be added manually
- pulling out and renaming the JUnit 4 test handler
- adding a new test handler for JUnit 5

I also added some changes to `.idea/` to facilitate developing with buck at addepar. At this point, addebuck is as tightly coupled with Addepar as it will ever be. 

Screenshot 2023-11-12 at 9 38 47 PM

@navkast @calcwu

### CODE REVIEW

**Comment by navkast** [comment]
surprisingly easy. can you swap it for 5.10.1 version of junit? should be a drop-in replacement.

**Comment by scottlee14** [comment]
Is there a major benefit to 5.10.1? It's pretty tedious to download JARs from maven for like 5 different dependencies. 

But I'll do it if there's a good reason.

**Comment by navkast** [comment]
bug fix around beforeall and beforeach between test inheritance across packages. it hurts us especially since we have cross-package test inheritance.

**Comment by scottlee14** [comment]
> bug fix around beforeall and beforeach between test inheritance across packages. it hurts us especially since we have cross-package test inheritance.

That's a good reason. Will do.

File: src/com/facebook/buck/testrunner/JUnitRunner.java [hunk 1]
```diff
@@ -86,25 +80,31 @@ public void run() throws Throwable {
...
-      if (mightBeATestClass(testClass)) {
+      if (mightBeJunit4TestClass(testClass)) {
         JUnitCore jUnitCore = new JUnitCore();
         Runner suite = new Computer().getSuite(createRunnerBuilder(), new Class<?>[] {testClass});
         Request request = Request.runner(suite);
         request = request.filterWith(filter);
-        jUnitCore.addListener(new TestListener(results, stdOutLogLevel, stdErrLogLevel));
+        jUnitCore.addListener(new Junit4TestListener(results, stdOutLogLevel, stdErrLogLevel, isDryRun));
         jUnitCore.run(request);
+      } else if (mightBeJunit5TestClass(testClass)) {
```
Comment (calcwu) [inline src/com/facebook/buck/testrunner/JUnitRunner.java:92]: This is nice. 👍

**Review by calcwu — COMMENTED** [review]
Awesome change - @scottlee-addepar 🚀

**Pull request merged**